### PR TITLE
fix #4837 (crash Blender if only first vertex in vertices list is empty)

### LINKED ISSUE #4837: Viewer Draw
## Problem statement

Viewer Draw node can crush Blender (at least in 3.4.0).

image

[untitled.blend.zip](https://github.com/nortikin/sverchok/files/10289095/untitled.blend.zip)

### PR DESCRIPTION
**Pull request by satabol**
Windows 11, Blender 3.4.1, Sverchok-master.

#4837 fix

### CODE REVIEW

File: nodes/viz/viewer_draw_mk4.py [hunk 1]
```diff
@@ -185,17 +185,20 @@ def view_3d_geom(context, args):
...
     if config.draw_verts:
-        bgl.glPointSize(config.point_size)
-        if config.uniform_verts:
-            v_batch = batch_for_shader(config.v_shader, 'POINTS', {"pos": geom.v_vertices})
-            config.v_shader.bind()
-            config.v_shader.uniform_float("color", config.vector_color[0][0])
-        else:
-            v_batch = batch_for_shader(config.v_shader, 'POINTS', {"pos": geom.v_vertices, "color": geom.points_color})
-            config.v_shader.bind()
+        if len(geom.v_vertices)>0 and (len(geom.v_vertices[0])==3) == True:
```
Comment (portnov) [inline nodes/viz/viewer_draw_mk4.py:188]: `(len() == 3) == True` looks odd. Why not just `len() == 3`?

Comment (satabol) [reply nodes/viz/viewer_draw_mk4.py:188]: To emphasize count of vertices. )
Should I skip this and remove “== True” ?

Comment (Durman) [reply nodes/viz/viewer_draw_mk4.py:188]: https://pep8.org/#programming-recommendations

image

Comment (satabol) [reply nodes/viz/viewer_draw_mk4.py]: @Durman  Is it good now? )

File: nodes/viz/viewer_draw_mk4.py [hunk 2]
```diff
@@ -185,17 +185,20 @@ def view_3d_geom(context, args):
...
+            else:
+                v_batch = batch_for_shader(config.v_shader, 'POINTS', {"pos": geom.v_vertices, "color": geom.points_color})
+                config.v_shader.bind()
 
-        v_batch.draw(config.v_shader)
-        bgl.glPointSize(1)
+            v_batch.draw(config.v_shader)
+            bgl.glPointSize(1)
+        else:
+            pass
```
Comment (Durman) [inline nodes/viz/viewer_draw_mk4.py:201]: Also this seems useless )

Comment (satabol) [reply nodes/viz/viewer_draw_mk4.py]: Fixed.

Comment (satabol) [reply nodes/viz/viewer_draw_mk4.py]: Also fix bgl.glDisable.

**Comment by satabol** [comment]
Fixed.
Also fix bgl.glDisable.

**Comment by satabol** [comment]
@Durman 
Fixed.
Also fix bgl.glDisable.

File: nodes/viz/viewer_draw_mk4.py [hunk 3]
```diff
@@ -185,19 +185,20 @@ def view_3d_geom(context, args):
...
-        bgl.glPointSize(config.point_size)
-        if config.uniform_verts:
-            v_batch = batch_for_shader(config.v_shader, 'POINTS', {"pos": geom.v_vertices})
-            config.v_shader.bind()
-            config.v_shader.uniform_float("color", config.vector_color[0][0])
-        else:
-            v_batch = batch_for_shader(config.v_shader, 'POINTS', {"pos": geom.v_vertices, "color": geom.points_color})
-            config.v_shader.bind()
+        if len(geom.v_vertices)>0 and (len(geom.v_vertices[0])==3):
+            bgl.glPointSize(config.point_size)
```
Comment (Durman) [inline nodes/viz/viewer_draw_mk4.py:189]: One more thing, if we are trying to bring all to pep8 =)

image

File: nodes/viz/viewer_draw_mk4.py [hunk 4]
```diff
@@ -185,19 +185,20 @@ def view_3d_geom(context, args):
...
+                v_batch = batch_for_shader(config.v_shader, 'POINTS', {"pos": geom.v_vertices, "color": geom.points_color})
+                config.v_shader.bind()
 
-        v_batch.draw(config.v_shader)
-        bgl.glPointSize(1)
+            v_batch.draw(config.v_shader)
+            bgl.glPointSize(1)
 
-    bgl.glEnable(bgl.GL_BLEND)
+    bgl.glDisable( bgl.GL_BLEND )
```
Comment (Durman) [inline nodes/viz/viewer_draw_mk4.py:202]: It's not important but just to let you know that it also contradicts with pep8 :p

image

File: nodes/viz/viewer_draw_mk4.py (hunk 3 above)
Comment (satabol) [reply nodes/viz/viewer_draw_mk4.py:189]: Sorry, I do not understand what you mean in this case? Should I remove something or check something else? )))

File: nodes/viz/viewer_draw_mk4.py (hunk 4 above)
Comment (satabol) [reply nodes/viz/viewer_draw_mk4.py:202]: I search code for “ .disable”. I’ll fix it.

File: nodes/viz/viewer_draw_mk4.py (hunk 3 above)
Comment (Durman) [reply nodes/viz/viewer_draw_mk4.py:189]:
```diff
- if len(geom.v_vertices)>0 and (len(geom.v_vertices[0])==3):
+ if geom.v_vertices and (len(geom.v_vertices[0])==3):
```

Comment (satabol) [reply nodes/viz/viewer_draw_mk4.py:189]: NO! This is source of problems! 
geom.v_vertices is True when len of geom.v_vertices == 0 ! So blender will crash on
geom.v_vertices[0]
before calc len(geom.v_vertices[0]) !!!

Comment (Durman) [reply nodes/viz/viewer_draw_mk4.py:189]: What I can see from the code is that `v_vertices` is a simple list. List, when it is empty, returns False. You can check with simple expression `bool([])`.

File: nodes/viz/viewer_draw_mk4.py (hunk 4 above)
Comment (satabol) [reply nodes/viz/viewer_draw_mk4.py]: done

File: nodes/viz/viewer_draw_mk4.py (hunk 3 above)
Comment (satabol) [reply nodes/viz/viewer_draw_mk4.py]: I never heard about this trick! Thank you. Fixed.

**Review by Durman — APPROVED** [review]

**Pull request merged**